src/coq_rules.ml

```diff
@@ -66,7 +71,9 @@ let setup_rule ~expander ~dir ~cc ~source_rule ~wrapper_name
   let stdout_to = Coq_module.obj_file ~obj_dir ~ext:".v.d" coq_module in
   let object_to = Coq_module.obj_file ~obj_dir ~ext:".vo"  coq_module in
 
-  let iflags = Arg_spec.As ["-R"; "."; wrapper_name] in
+  let src_dir = Path.drop_build_context_exn dir in
+  let th_flags = List.concat_map theories_libs ~f:setup_theory_flag in
+  let iflags = Arg_spec.As (["-R"; Path.to_string src_dir; wrapper_name] @ th_flags) in
```

Comment:
I believe you can use the `Path` constructor in `Arg_spec` in conjunction with `S`. to avoid the path stringification.

Reply:
Done, thanks; improved a bit the general flag setup; still can be better tho.

Reply:
Umm, I had to revert the use of `Arg_spec.Path` as it was producing wrong paths to `-Q`, instead of `-Q ssreflect` I am getting `-Q ../../ssreflect`. I will investigate more.

Reply:
Oh Indeed, it was my fault, `Path` handles `Build.run` `~dir` properly so I had to drop the drop_build_context hack. At some point I was aware of this but I forgot. Path code looks clean now IMO.